api/tests/Behat/bootstrap/v2/ClientManagement/ClientManagementTrait.php

```diff
@@ -369,4 +369,58 @@ public function theClientDoesNotHaveANamedDeputyAssociatedWithThem()
         $this->em->persist($client);
         $this->em->flush();
     }
+
+    /**
+     * @When I attempt to unarchived the client
+     */
+    public function iAttemptToUnarchivedTheClient()
+    {
+        $this->assertInteractingWithUserIsSet();
+
+        try {
+            $this->clickLink('Un-archive client');
+            $this->iAmOnAdminClientUnarchivedPage();
+            $this->clickLink('Return to client dashboard');
+        } catch (\Throwable $e) {
+            // This step is used as part of testing the unarchive button isnt here so swallow errors and assert on following step
+        }
+    }
+
+    /**
+     * @When an org deputy has an archived client
+     */
+    public function theDeputyHasAnArchivedClient()
+    {
+        $clientId = $this->interactingWithUserDetails->getClientId();
```

Comment:
To make the use of this step more friendly we probably want to call `$this->assertInteractingWithUserIsSet();` here too so any future users of the code will get a useful failure error is the user details aren't set.